jira/resources.py

```diff
@@ -482,6 +482,13 @@ def permalink(self):
     def __eq__(self, other):
         return self.id == other.id
 
+    def __getitem__(self, item):
+        # Grab the JIRA server
+        jira_server = self._session.jira_server
+
+        # Look for the appropriate fields list on the server
+        return getattr(self.fields, jira_server._fields[item])
```

Comment:
Not sure duplicates are too much of a concern (this would only arise if two custom fields differ in name only by punctuation or spacing, which seems pretty edge-case).

It would be good to turn the custom field names into nice python identifiers though. How about this:
* Replace spaces with underscores
* Strip invalid characters
* Lowercase the rest

"Requestor's Manager" becomes "requesters_manager".

I figured this was a solved problem, but after some googling didn't turn up anything that quite fits this, but it shouldn't be too much trouble to hand-roll something appropriate.